longpoll-user/v10/wrapper.go

```diff
@@ -0,0 +1,408 @@
+		event := ChatParamsChange{}
+		if err := event.Parse(i); err != nil {
+			return err
+		}
+		f(event)
+
+		return nil
+	})
+}
+
+// ChatInfoChangeHandler handler func for ChatInfoChange
+type ChatInfoChangeHandler func(m ChatInfoChange)
+
+// OnChatInfoChange func
+//
+// event with code 52
+//
+// TODO: write description
+func (w Wrapper) OnChatInfoChange(f ChatInfoChangeHandler) {
+	w.longpoll.EventNew(52, func(i []interface{}) error {
+		event := ChatInfoChange{}
+		if err := event.Parse(i); err != nil {
+			return err
+		}
+		f(event)
+
+		return nil
+	})
+}
+
+// UsersTypingHandler handler func for UsersTyping
+type UsersTypingHandler func(m UsersTyping)
+
+// OnUsersTyping func
+//
+// event with code 63
+//
+// TODO: write description
+func (w Wrapper) OnUsersTyping(f UsersTypingHandler) {
+	w.longpoll.EventNew(63, func(i []interface{}) error {
+		event := UsersTyping{}
+		if err := event.Parse(i); err != nil {
+			return err
+		}
+		f(event)
+
+		return nil
+	})
+}
+
+// UsersRecordingAudioMessageHandler handler func for UsersRecordingAudioMessage
+type UsersRecordingAudioMessageHandler func(m UsersRecordingAudioMessage)
+
+// OnUsersRecordingAudioMessage func
+//
+// event with code 64
+//
+// TODO: write description
+func (w Wrapper) OnUsersRecordingAudioMessage(f UsersRecordingAudioMessageHandler) {
+	w.longpoll.EventNew(64, func(i []interface{}) error {
+		event := UsersRecordingAudioMessage{}
+		if err := event.Parse(i); err != nil {
+			return err
+		}
+		f(event)
+
+		return nil
+	})
+}
+
+// CounterChangeHandler handler func for CounterChange
+type CounterChangeHandler func(m CounterChange)
+
+// OnCounterChange func
+//
+// event with code 80
+//
+// TODO: write description
+func (w Wrapper) OnCounterChange(f CounterChangeHandler) {
+	w.longpoll.EventNew(80, func(i []interface{}) error {
+		event := CounterChange{}
+		if err := event.Parse(i); err != nil {
+			return err
+		}
+		f(event)
+
+		return nil
+	})
+}
+
+// FriendInvisibilityChangeHandler handler func for FriendInvisibilityChange
+type FriendInvisibilityChangeHandler func(m FriendInvisibilityChange)
+
+// OnFriendInvisibilityChange func
+//
+// event with code 81
+//
+// TODO: write description
+func (w Wrapper) OnFriendInvisibilityChange(f FriendInvisibilityChangeHandler) {
+	w.longpoll.EventNew(81, func(i []interface{}) error {
+		event := FriendInvisibilityChange{}
+		if err := event.Parse(i); err != nil {
+			return err
+		}
+		f(event)
+
+		return nil
+	})
+}
+
+// NotificationSettingsChangeHandler handler func for NotificationSettingsChange
+type NotificationSettingsChangeHandler func(m NotificationSettingsChange)
+
+// OnNotificationSettingsChange func
+//
+// event with code 114
+//
+// TODO: write description
+func (w Wrapper) OnNotificationSettingsChange(f NotificationSettingsChangeHandler) {
+	w.longpoll.EventNew(114, func(i []interface{}) error {
+		event := NotificationSettingsChange{}
+		if err := event.Parse(i); err != nil {
+			return err
+		}
```

Comment:
При `mode=10`

>  cannot parse array into Go struct NotificationSettingsChange: expected at least 3 element(s), got 2

```suggestion
		event := NotificationSettingsChange{}
		if w.longpoll.Mode & longpoll.ExtendedEvents != 0 {
			if err := event.ParseMode8(i); err != nil {
				return err
			}
		} else {
			if err := event.Parse(i); err != nil {
				return err
			}
		}
```

Тут нужно еще добавить в импорт `longpoll`